pkg/services/scaling/metadata/cluster.go

```diff
@@ -15,8 +15,18 @@ func GetClusterInfo() (projectId, zone string, err error) {
 	zoneReq, err := http.NewRequest("GET", "http://metadata.google.internal/computeMetadata/v1/instance/zone", nil)
 	zoneReq.Header.Add("Metadata-Flavor", "Google")
 	zoneResp, err := httpclient.Do(zoneReq)
-	defer zoneResp.Body.Close()
-	defer projectIdResp.Body.Close()
+	defer func() {
+		err = zoneResp.Body.Close()
+	}()
+	if err != nil {
+		return "", "", err
+	}
+	defer func() {
+		err = projectIdResp.Body.Close()
+	}()
+	if err != nil {
+		return "", "", err
+	}
```

Comment:
the err check here happens after the defer statement?

i think we could just defer the close on both these without error checking?
do we really care if we processed everything?

Reply:
one of the linters requires all errors to be checked which leads to weird syntax. I could disable the err check linter for this bit

Reply:
i think we can use `// nolint:errcheck`

Reply:
```suggestion
	defer zoneResp.Body.Close() // nolint:errcheck
	defer projectIdResp.Body.Close() // nolint:errcheck
```